### ISSUE TITLE
Extended API usage

### ISSUE THREAD

**Issue by erasche** [erasche/argparse2tool#10]
# Conditionals

Need some way to support conditionals.

``` python
parser = argparse.ArgumentParser()
parser.add_argument('bar', choices=('a', 'b', 'c'))

def bar_test(value):
    return value == 'a'

parser.add_argument('baz', depends_on='bar', when=bar_test)
```
# ????

Please comment/edit issue for input.

**Comment by TorHou** [comment]
### subparser

Two people in our group immediately after using argparse started using subparser. This would be really nice to have, yet I guess it might be tricky to implement.

As far as I understand they use subparser for mode setting. Different modes enable different parameters and different `usage` texts are displayed depending on the mode. So this comes back to conditionals in Galaxy I guess.

**Comment by erasche** [comment]
Makes sense @TorHou :) It shouldn't be too tough to wrap the whole thing in a giant conditional for all subparser blocks
